toucan_connectors/snowflake_common.py

```diff
@@ -147,12 +148,15 @@ def add_default_order_if_needed(
         query: str,
         query_columns: Optional[List] = None,
     ):
-        """If it's a SELECT type query and there is not an 'ORDER BY' inside, we sort with the first column of the
-        dataset"""
+        """If it's a SELECT type query and there is not an 'ORDER BY' inside, we sort using ROW_NUMBER function"""
 
-        if query_columns and 'WITH SELECT_STEP_' in query and ' ORDER BY ' not in query:
+        if (
+            query_columns
+            and 'SELECT' in query.upper()
```

Comment:
:+1: Great, thanks!
Maybe we should also update the tests so that they contain SELECT and not WITH SELECT_STEP?

Reply:
Updated !